composer/lib/dependabot/composer/file_parser.rb

```diff
@@ -165,9 +165,18 @@ def git_dependency_details(package_details, requirement)
       end
 
       def lockfile_details(name:, type:)
-        key = lockfile_key(type)
-        parsed_lockfile.fetch(key, []).find { |d| d["name"] == name }
-      end
+		begin
+           key = lockfile_key(type) 
```

Comment:
do we know why this can throw?

Reply:
well, what I found is that the file_parser is actually looking at all files in a folder.  It ignores the name of the file. when looking for the key fields.  My thought is that the reason we are seeing the issue is that we are not filtering the filees we parse based on name or extension.  If we did that, it would limit us.  so, type could be nil, so could key, parsed_lockfile, parsed_lockfile.fetch(key,[])  or d.  The problem is that we are not getting the nil in lockfile because any file can be parsed - just not properly.